cmdeploy/src/cmdeploy/__init__.py

```diff
@@ -375,17 +385,26 @@ def _configure_dovecot(config: Config, debug: bool = False) -> bool:
     return need_restart
 
 
-def _configure_nginx(domain: str, debug: bool = False) -> bool:
+def _configure_nginx(domain: str, debug: bool = False, config = Config) -> bool:
```

Comment:
I'm not sure what I have to do here to pass config into this function. Adding it like that leads to `AttributeError: 'str' object has no attribute 'disable_ipv6'`

Reply:
@missytake any ideas?

Reply:
First of all, I think you mean `config: Config` not `config = Config`; the former is specifying a type, the latter is setting a default value :)

And you need to pass the config to the `_configure_nginx()` function, afaict that doesn't happen yet? So I wonder why `config` is a string object instead of a class object here, that doesn't make much sense.

Reply:
In general you can take a look at https://github.com/deltachat/chatmail/pull/285/files and how I added a config value there, I think there it worked

Reply:
Problem is, I tried adding config: Config first and that leads to `TypeError: _configure_nginx() missing 1 required positional argument: 'domain'`.
I did it the same way as in the PR you linked, and it works in the other functions in __init__.py.

I can't find any other mention of the _configure_nginx() function in the entire project, so I don't get where it should get this value.

Reply:
It is called here, with a string: https://github.com/deltachat/chatmail/blob/hagi/%23306-disable-ipv6/cmdeploy/src/cmdeploy/__init__.py#L595
you should pass the whole config here, instead of only the domain. Luckily it's really only called once.